lock-api/src/main/java/com/palantir/lock/client/RemoteTimelockServiceAdapter.java

```diff
@@ -173,20 +195,20 @@ public long currentTimeMillis() {
 
     @Override
     public TimestampLeaseResults acquireTimestampLeases(Map<TimestampLeaseName, Integer> requests) {
-        // TODO(aalouane): implement!
-        throw new UnsupportedOperationException("Not implemented yet!");
+        return timestampLeaseAcquirer.acquireNamedTimestampLeases(requests);
     }
 
     @Override
     public Map<TimestampLeaseName, Long> getMinLeasedTimestamps(Set<TimestampLeaseName> timestampNames) {
-        // TODO(aalouane): implement!
-        throw new UnsupportedOperationException("Not implemented yet!");
+        return minLeasedTimestampGetter.getMinLeasedTimestamps(timestampNames);
     }
 
     @Override
     public void close() {
         transactionStarter.close();
         commitTimestampGetter.close();
         lockLeaseService.close();
+        timestampLeaseAcquirer.close();
```

Comment:
Can we FLUP later on wrapping this in some sort of closer. ANy exception thrown here (if someone does something wrong, will cause us to leak

Reply:
Tracked